### PR TITLE
Eslint

### PR DESCRIPTION
**Pull request by godu**

### CODE REVIEW

File: packages/@coorpacademy-components/src/atom/autocomplete/index.js [hunk 1]
```diff
@@ -22,17 +21,22 @@ const Autocomplete = props => {
...
-  const handleSuggestionsSelected = (e, data) => onSuggestionSelected(data);
+  const handleChange = useMemo(() => e => onChange(e), [onChange]);
+  const handleBlur = useMemo(() => (e, selectedSuggestion) => onBlur(e, selectedSuggestion), [
+    onBlur
+  ]);
+  const handleSuggestionsFetchRequested = useMemo(() => e => onFetch(e), [onFetch]);
+  const handleSuggestionsClearRequested = useMemo(() => e => onClear(e), [onClear]);
+  const handleSuggestionsSelected = useMemo(() => (e, data) => onSuggestionSelected(data), [
+    onSuggestionSelected
+  ]);
```
Comment (Fausto95) [inline packages/@coorpacademy-components/src/atom/autocomplete/index.js:39]:
```suggestion
  const handleSuggestionsSelected = useMemo(() => (e, data) => onSuggestionSelected(data), [
    onSuggestionSelected,
    data
  ]);
```

Applicable everywhere, and please install https://www.npmjs.com/package/eslint-plugin-react-hooks so we get warned

File: packages/@coorpacademy-components/src/atom/cta/index.js [hunk 2]
```diff
@@ -18,13 +35,13 @@ class CTA extends React.Component {
   }
 
   handleMouseEnter() {
-    return this.setState(prevState => ({
+    return this.setState(() => ({
```
Comment (Fausto95) [inline packages/@coorpacademy-components/src/atom/cta/index.js:38]: Why even use an updater if we're not depending on previous state, just pass an object

File: packages/@coorpacademy-components/src/atom/autocomplete/index.js (hunk 1 above)
Comment (godu) [reply packages/@coorpacademy-components/src/atom/autocomplete/index.js:39]: `data` is not in this scope. You misread the code.

Yeah, we could add `eslint-plugin-react-hooks`. You could do a PR on https://github.com/CoorpAcademy/eslint-plugin-coorpacademy for that.

File: packages/@coorpacademy-components/src/atom/cta/index.js (hunk 2 above)
Comment (godu) [reply packages/@coorpacademy-components/src/atom/cta/index.js:38]: I removed unused params. I could do more, but that you propose is not iso. There is a subtle difference between both. Mine is lazy, your's eager.
setState wait in the first place an updater function.
In this case, we don't care, the state isn't too complexe to pay attention of evaluation's performance.

File: packages/@coorpacademy-components/src/atom/autocomplete/index.js (hunk 1 above)
Comment (Fausto95) [reply packages/@coorpacademy-components/src/atom/autocomplete/index.js:39]: Oh yea

File: packages/@coorpacademy-components/src/atom/input-html/index.js [hunk 3]
```diff
@@ -26,31 +36,37 @@ class InputHtml extends React.Component {
...
-    };
-    const handlePreview = e => {
-      e.preventDefault();
-      this.setState(state => ({
-        text: state.text,
-        preview: !state.preview
-      }));
-    };
-    const iconContent = !this.state.preview ? (
+    const iconContent = !preview ? (
```
Comment (silouone) [inline packages/@coorpacademy-components/src/atom/input-html/index.js:69]: pk le ternaire est construit ds ce sens la :p ?

Comment (godu) [reply packages/@coorpacademy-components/src/atom/input-html/index.js:69]: Aucune idée. Mais c'était déjà le cas avant.

**Pull request merged**